lib/git.zsh

```diff
@@ -12,11 +12,53 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+  local branches
+  local info
+  local deatchedtag
+
+  # If oh-my-zsh.hide-status equals to '1', get out.
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" == "1" ]]; then
+    return 0
+  fi
+
+  # If we are on a folder not tracked by git, get out.
+  if [[ "$(__git_prompt_git rev-parse --git-dir 2> /dev/null)" == "" ]]; then
+    return 0
+  fi
+
+  # If oh-my-zsh.classic-info is set, use classic info output of branch only
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.classic-info 2>/dev/null)" == "1" ]]; then
```

Comment:
I don't like that you have to opt-in to the old behavior, it should be other way around. Also, putting this under another git config option isn't ideal, I'd prefer a `ZSH_GIT_PROMPT_*` variable.

Reply:
There are two things to discuss here:

1. Opt-out: I think you are right that it is better to have this change as `opt-in`. Well documented and advertised in the release notes, so people can give it a try. Later on this could change to an `opt-out`, if the feedback is positive enough.

2. git config: The reason I went with a `git config` option is to allow the user to switch this feature _per repository_ if they want. You can have it on globally, but off on specific repositories. I found this useful for some _very large_ repositories (e.g. github.com/openjdk/jdk).

All said, I believe we can go with `opt-in` and the git config option. What do you think @mcornella ?